webviz_subsurface/plugins/_multiple_regression.py

```diff
@@ -0,0 +1,1129 @@
+            2.  Create a model matrix, fit the model and calculate selection criterion for each
+                remaining parameter.
+            3.  Pick the best parameter and repeat with remaining parameters until we satisfy an
+                exit condition.
+            4.  Finally fit a Statsmodels regression and return the results.
+
+        Exit conditions:
+            - No parameters in remaining.
+            - The last model was not the best model.
+            - Hit cap on maximum parameters.
+            - We are about to add more parameters than there are observations.
+     """
+
+    # Initialize values for use in algorithm (sst is the total sum of squares)
+    response = data[resp].to_numpy(dtype="float32")
+    # Check for constant response
+    if np.all(response == response[0]):
+        return None
+    sst = np.sum((response - np.mean(response)) ** 2)
+    remaining = set(data.columns).difference(set(force_in + [resp]))
+    selected = force_in
+    current_score, best_new_score = 0.0, 0.0
+    while remaining and current_score == best_new_score and len(selected) < maxvars:
+        scores_with_candidates = []
+        for candidate in remaining:
+            if " × " in candidate:
+                current_model = (
+                    selected.copy()
+                    + [candidate]
+                    + list(set(candidate.split(" × ")).difference(set(selected)))
+                )
+            else:
+                current_model = selected.copy() + [candidate]
+            parameters = data.filter(items=current_model).to_numpy(dtype="float64")
+            num_parameters = parameters.shape[1]
+            parameters = np.append(parameters, np.ones((len(response), 1)), axis=1)
+
+            # Fit model
+            try:
+                beta = la.inv(parameters.T @ parameters) @ parameters.T @ response
+            except la.LinAlgError:
+                # This clause lets us skip singluar and other non-valid model matricies.
+                continue
+
+            if len(response) - num_parameters - 1 < 1:
+                # The exit condition means adding this parameter would add more parameters than
+                # observations. This causes infinite variance in the model so we return the current
+                # best model
+
+                model_df = data.filter(items=selected)
+                model_df["Intercept"] = np.ones((len(response), 1))
+                model_df["response"] = response
+
+                return _model_warnings(model_df)
+
+            f_vec = beta @ parameters.T
+            ss_res = np.sum((f_vec - np.mean(response)) ** 2)
+
+            r_2_adj = 1 - (1 - (ss_res / sst)) * (
+                (len(response) - 1) / (len(response) - num_parameters - 1)
+            )
+            scores_with_candidates.append((r_2_adj, candidate))
+
+        # If the best parameter is interactive, add all base features
```

Comment:
A bit of mixing between the terms feature and parameter here. In addition, it should be interaction, not interactive.